includes/manifest.inc

```diff
@@ -7,98 +7,131 @@
-    islandora_iiif_manifests_create_json_response('403 Forbidden', 'Forbidden');
+    if (!isset($object[ISLANDORA_IIIF_MANIFESTS_DATASTREAM_ID])) {
+      $response = array(
+        "data" => "No manifest found",
+        "status" => "400 Bad request"
+      );
+    }
+    if (islandora_datastream_access(ISLANDORA_VIEW_OBJECTS, $object[ISLANDORA_IIIF_MANIFESTS_DATASTREAM_ID])) {
+      $response = array(
+        "data" => "Forbidden",
+        "status" => "403 Forbidden"
+      );
+    }
+    islandora_iiif_manifests_create_json_response($response['status'], $response['data']);
   }
+  
 }
 
 /**
  * Response in JSON for the manifest.
  */
-function islandora_iiif_manifests_create_manifest_json(AbstractObject $object) {
-
+function islandora_iiif_manifests_create_manifest_json(AbstractObject $object, $collection = NULL) {
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
-
-  $output = islandora_iiif_manifests_build_manifest($object);
-
-  islandora_iiif_manifests_create_json_response('200 OK', $output);
+  
+  $output = islandora_iiif_manifests_build_manifest($object,$collection);
+  
+  if (!$output) {
+    $response = array(
+      "data" => "No manifest found in collection",
+      "status" => "400 Bad request"
+    );
+    islandora_iiif_manifests_create_json_response($response['status'], $response['data']);
+  }
+  else
+    islandora_iiif_manifests_create_json_response('200 OK', $output);
 }
 
+
 /**
- * Response in JSON for the manifest.
+ * Create JSON manifest.
  */
 function islandora_iiif_manifests_create_manifest_datastream(AbstractObject $object) {
-
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
-
+  
   $output = islandora_iiif_manifests_build_manifest($object);
-
+  
   islandora_iiif_manifests_add_manifest_datastream($object, $output);
 }
 
 /**
  * Response in JSON for the canvas.
  */
 function islandora_iiif_manifests_create_canvas_json(AbstractObject $object) {
-
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
-
+  
   $output = islandora_iiif_manifests_build_canvas($object);
-
+  
   islandora_iiif_manifests_create_json_response('200 OK', $output);
 }
 
 /**
  * Build manifest array.
  */
-function islandora_iiif_manifests_build_manifest($object) {
-
+function islandora_iiif_manifests_build_manifest($object, $collection = NULL ) {
+  
   $output = array();
-
+  
   $hooks = islandora_build_hook_list(ISLANDORA_IIIF_MANIFESTS_ISLANDORA_IIIF_MANIFEST_HOOK, $object->models);
-
+  
   foreach ($hooks as $hook) {
-    $temp = module_invoke_all($hook, $object);
-
+    $temp = module_invoke_all($hook, $object, $collection);
+    
     if (!empty($temp)) {
       $output = array_merge_recursive($output, $temp);
     }
   }
-
-  return json_encode($output, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE | JSON_NUMERIC_CHECK);
+  
+  if ($output['@type'] == "sc:Collection" && !isset($output['members']) && !isset($output['collections'])) {
```

Comment:
Upon reading IIIF the documentation, it is better to change the response to return an empty collection manifest. This would be better for IIIF viewers to process and could avoid console errors, such as bellow.

PS: "At least one of collections, manifests and members should be present in the response. An empty collection, with no member resources, is allowed but discouraged." We are using [https://iiif.io/api/presentation/2.1/#collection](url)  and not [https://iiif.io/api/presentation/3.0/#51-collection](url)

![image](https://user-images.githubusercontent.com/50174699/112465510-1b0d4580-8d65-11eb-8122-579b570b1e51.png)

Reply:
My previous comment was too soon...  but I agree!